### PR TITLE
Round the epilogue offset to positive values in cunn_SoftMaxForward

### PR DESCRIPTION
**Pull request by xinyazhang**
This fixes OOB memory access for followng code
``` python
import torch
qk = torch.randn((9,1017), dtype=torch.float64, device='cuda')
smqk = torch.softmax(qk, dim=-1)
```

Correctness can be confirmed with:
``` python
import torch
import numpy as np
from scipy.special import softmax

qk = torch.randn((9,1017), dtype=torch.float64, device='cuda')
nqk = qk.cpu().numpy()
smqk = torch.softmax(qk, dim=-1)
nsmqk = smqk.cpu().numpy()
smnqk = softmax(nqk, axis=-1)

print(f'{np.allclose(smnqk, nsmqk)}')
```

### CODE REVIEW

**Comment by ngimel** [comment]
Still a no
```
In [1]: import torch
   ...: M = 9
   ...: N = 1017
   ...: dtype = torch.float64
   ...:
   ...: def test_softmax_oob_access():
   ...:     qk_with_margin = torch.randn((M+2, N), dtype=dtype, device="cuda");
   ...:     qk_with_margin[-1].fill_(float('nan'))
   ...:     qk_with_margin[0].fill_(float('nan'))
   ...:     qk = qk_with_margin[1:-1, :]
   ...:     smqk = torch.softmax(qk, dim=-1)
   ...:     assert not torch.isnan(smqk).any(), 'NaN indicates OOB memory access'
   ...:
   ...: test_softmax_oob_access()

In [2]: test_softmax_oob_access()
```

**Comment by xinyazhang** [comment]
> Still a no

Apparently cannot be reproduced on H100. The error occurs on V100/MI
Speculating H100 is not calling this kernel.

**Comment by ngimel** [comment]
There's no hardware-specific dispatch in softmax. Which kernel are you calling? Why is offset less than 0? The way the function is written it should never be less than 0.

**Comment by ngimel** [comment]
The only place where it does `size -= blockDim.x;` is in writeBpropResultsVectorized where it indeed may be a bug?, in other places it's `size -= blockDim.x > size ? size : blockDim.x;` so size is always non-negative.
